File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public Integer[] solution(int[] array, int[][] commands) {
        
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < commands.length; i++) {
            List<Integer> list = new ArrayList<>();
            for (int j = commands[i][0] - 1; j < commands[i][1]; j++) {
                list.add(array[j]);
            }
            Collections.sort(list);
            ans.add(list.get(commands[i][2] - 1));
        }
        Integer[] answer = ans.toArray(new Integer[0]);
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for (vector<int> c : commands) {
        vector<int> sub_array(array.begin() + c[0] - 1, array.begin() + c[1]);
        sort(sub_array.begin(), sub_array.end());
        answer.push_back(sub_array[c[2] - 1]);
        
    }
    return answer;
}